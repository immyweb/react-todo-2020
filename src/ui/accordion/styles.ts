import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
