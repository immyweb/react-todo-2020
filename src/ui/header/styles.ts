import styled from '@emotion/styled';

export const HeaderContainer = styled.header`
  margin-bottom: 25px;
`;

export const Heading = styled.h1`
  margin: 0;
`;
