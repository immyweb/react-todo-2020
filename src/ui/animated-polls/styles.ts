import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { fonts } from '../../styles/fonts';
import { H2, P } from '../typography/typography';

// Typography

export const APHeadline = glamorous(H2)({
  label: 'AP-headline',
  fontSize: 42,
  textTransform: 'uppercase',
  textAlign: 'center',
  color: colours.$white,
  [mediaQueries.tabletPortrait]: {
    fontSize: 60,
    lineHeight: 1,
  },
});

export const APCopy = glamorous(P)({
  label: 'AP-copy',
  fontSize: 16,
  textAlign: 'center',
  color: colours.$white,
  lineHeight: 1.13,
  marginBottom: 25,
  [mediaQueries.tabletPortrait]: {
    fontSize: 24,
  },
});

// Animated Polls

export const APoll = glamorous.section({
  label: 'AP',
  position: 'relative',
});

// Modal

export const ModalContainer = glamorous.div<{ open: boolean }>(
  {
    label: 'modal-container',
    position: 'fixed',
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    zIndex: 160,
    [mediaQueries.tabletPortrait]: {
      position: 'absolute',
      height: 900,
      zIndex: 40,
    },
  },
  ({ open }) => ({
    visibility: open ? 'visible' : 'hidden',
  }),
);

export const ModalInner = glamorous.div({
  label: 'modal-inner',
  position: 'relative',
  height: '100vh',
  width: '100%',
  top: 0,
  background: colours.$black,
  overflowY: 'hidden',
  msOverflowStyle: '-ms-autohiding-scrollbar',
  [mediaQueries.tabletPortrait]: {
    height: 900,
  },
});

export const ModalContent = glamorous.div({
  label: 'modal-content',
  position: 'relative',
  overflowY: 'auto',
  msOverflowStyle: '-ms-autohiding-scrollbar',
  height: '100%',
});

export const ModalHeadline = glamorous.h2({
  label: 'modal-headline',
  textTransform: 'uppercase',
  textAlign: 'center',
  fontSize: 24,
  fontFamily: fonts.$theSunHeavyCondensed,
  marginBottom: 20,
  lineHeight: 1.1,
  padding: '0 25px',
  [mediaQueries.tabletPortrait]: {
    padding: 0,
  },
});

export const BtnClose = glamorous.div({
  label: 'modal-btn-close',
  textAlign: 'center',
  position: 'absolute',
  top: 23,
  right: 16,
  height: 28,
  background: 'none',
  borderRadius: '50%',
  ':hover': {
    cursor: 'pointer',
  },
  [mediaQueries.tabletPortrait]: {
    top: 35,
    right: 35,
  },
});

// Teaser screen

export const APTeaser = glamorous.section({
  label: 'AP-teaser',
  width: '100%',
  position: 'relative',
  minHeight: 520,
  [mediaQueries.tabletPortrait]: {
    height: 900,
    overflow: 'hidden',
  },
});

export const APTeaserLogo = glamorous.img<{ backgroundColour: string }>(
  {
    label: 'AP-teaser-logo',
    position: 'absolute',
    zIndex: 3,
    top: 8,
    left: 8,
    width: 72,
    padding: '8px 4px',
    [mediaQueries.mobileInner]: {
      top: 20,
      left: 20,
    },
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const APTeaserImage = glamorous.div({
  label: 'AP-teaser-image',
  width: '100%',
  position: 'absolute',
  bottom: 0,
  left: 0,
  height: '100%',
  backgroundBlendMode: 'multiply',
  backgroundSize: 'cover',
  backgroundPosition: 'top center',
  backgroundRepeat: 'no-repeat',
  backgroundImage: `linear-gradient(to bottom, ${
    colours.$black
  } 0%, transparent 20%, transparent 50%, ${colours.$black} 81%, ${colours.$black} 100%)`,
  [mediaQueries.mobileInner]: {
    backgroundImage: `linear-gradient(to bottom, ${
      colours.$black
    } 0%, transparent 20%, transparent 50%, ${colours.$black} 81%, ${colours.$black} 100%)`,
  },
});

export const APTeaserVideo = glamorous.video({
  label: 'AP-teaser-video',
  width: '100%',
});

export const APTeaserInner = glamorous.div({
  label: 'AP-teaser-inner',
  position: 'absolute',
  bottom: 20,
  margin: '0 40px',
  [mediaQueries.mobileInner]: {
    bottom: 40,
    margin: '0 100px',
  },
});

// Intro Screen

export const APIntro = glamorous.section({
  label: 'AP-intro',
  width: '100%',
  position: 'relative',
  height: '100vh',
  background: colours.$black,
  [mediaQueries.mobileInner]: {
    height: 900,
    background: 'none',
  },
});

export const APIntroImg = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'AP-intro-img',
    position: 'absolute',
    width: '100%',
    top: 0,
    left: 0,
    height: 520,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
      height: 900,
    },
  }),
);

export const APIntroCopy = glamorous.div({
  label: 'AP-intro-copy',
  position: 'absolute',
  width: '100%',
  bottom: 0,
  padding: '0 40px 30px',
  background: colours.$black,
  [mediaQueries.mobileInner]: {
    paddingBottom: 15,
    background: 'none',
  },
});

export const APIntroLink = glamorous.a({
  label: 'AP-intro-link',
  color: colours.$midGrey,
  textDecoration: 'none',
});

export const APIntroHashTag = glamorous.span({
  label: 'AP-intro-hashtag',
  color: colours.$white,
  letterSpacing: 1.4,
});

// Card Screen

export const APCard = glamorous(APIntro)({
  label: 'AP-card',
});

export const APCardImg = glamorous(APIntroImg)({
  label: 'AP-card-image',
  height: '100vh',
  [mediaQueries.mobileInner]: {
    height: 900,
  },
});

export const APCardCopy = glamorous(APIntroCopy)({
  label: 'AP-card-copy',
  background: colours.$black,
  paddingTop: 15,
  minHeight: 165,
  ':before': {
    content: ' ',
    position: 'absolute',
    width: '100%',
    left: 0,
    top: -100,
    height: 100,
    backgroundImage: `linear-gradient(to bottom, rgba(0, 0, 0, 0), ${colours.$black})`,
  },
  [mediaQueries.mobileInner]: {
    background: colours.$black,
    minHeight: 'initial',
  },
});

// Vote Screen

export const APVote = glamorous(APIntro)({
  label: 'AP-vote',
  marginBottom: 0,
});

export const APVoteImgBkgnd = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'AP-vote-img-bkgnd',
    position: 'absolute',
    width: '100%',
    top: 0,
    left: 0,
    height: '100vh',
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const APVoteImgBkgndCover = glamorous.div({
  label: 'AP-vote-img-bkgnd-cover',
  position: 'absolute',
  top: 0,
  left: 0,
  width: '100%',
  height: '100vh',
  background: colours.$black,
  opacity: 0.8,
});

export const APVoteImgBkgndCoverGradient = glamorous.div({
  label: 'AP-vote-img-bkgnd-cover-gradient',
  position: 'absolute',
  left: 0,
  bottom: 0,
  width: '100%',
  height: '60%',
  background: `linear-gradient(to bottom, rgba(255,255,255,0) 0%,rgba(0,0,0,1) 100%)`,
});

export const APVoteHeadline = glamorous(APHeadline)({
  label: 'AP-vote-headline',
  textTransform: 'none',
  textAlign: 'left',
  fontSize: 38,
});

export const APVoteContent = glamorous.div({
  label: 'AP-vote-content',
  width: '100%',
  height: '100vh',
  position: 'absolute',
  top: 0,
  left: 0,
  padding: '90px 20px 0',
});

export const APVoteSubHeadline = glamorous(APVoteHeadline)({
  label: 'AP-vote-sub-headline',
  [mediaQueries.mobileInner]: {
    textAlign: 'center',
    fontSize: 40,
    padding: '25px 0',
  },
});

// Option bar

export const APOptionBar = glamorous.div({
  label: 'AP-option-bar',
  position: 'relative',
  marginBottom: 12,
  display: 'flex',
});

export const APOptionBarLabel = glamorous.div<{ bkgnd: string }>(
  {
    label: 'AP-option-bar-label',
    display: 'flex',
    alignItems: 'center',
    height: 44,
    width: '100%',
    paddingLeft: 8,
  },
  ({ bkgnd }) => ({
    background: bkgnd,
  }),
);

export const APOptionBarIcon = glamorous.img({
  label: 'AP-option-bar-icon',
  width: 35,
  display: 'inline-block',
  marginRight: 20,
});

export const APOptionBarIconLabel = glamorous(P)({
  label: 'AP-option-bar-icon-label',
  display: 'inline-block',
  color: colours.$white,
  marginBottom: 0,
});

export const APOptionBarCta = glamorous.div<{ bkgnd: string }>(
  {
    label: 'AP-option-bar-cta',
    display: 'inline-block',
    width: 52,
    height: 44,
    marginLeft: 5,
  },
  ({ bkgnd }) => ({
    background: bkgnd,
  }),
);

export const APOptionBarCtaIcon = glamorous.span({
  label: 'AP-option-bar-cta-icon',
  display: 'inline-block',
  position: 'relative',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
});

// Results Screen

export const APResultsHeadline = glamorous(APVoteHeadline)({
  label: 'AP-results-headline',
  textTransform: 'none',
  textAlign: 'center',
  fontSize: 38,
  [mediaQueries.mobileInner]: {
    fontSize: 40,
    padding: '25px 0',
  },
});

export const APChartHolder = glamorous.div({
  label: 'AP-option-bar-cta-icon',
  width: '100%',
  height: 330,
  marginBottom: 10,
});

// Progress bar

export const APProgressBar = glamorous.div({
  label: 'AP-progress-bar',
  width: '100%',
  display: 'flex',
  justifyContent: 'space-around',
  padding: '0 13px',
  [mediaQueries.tabletPortrait]: {
    padding: '0 25px',
  },
});

export const APProgressBarItem = glamorous.div<{ viewed: boolean }>(
  {
    label: 'AP-progress-bar-item',
    background: colours.$white,
    height: 4,
    margin: '0 3px',
    flexGrow: 1,
    [mediaQueries.tabletPortrait]: {
      height: 6,
    },
  },
  ({ viewed }) => ({
    opacity: viewed ? 1.0 : 0.5,
  }),
);

// Navigation

export const APNav = glamorous.div({
  label: 'AP-nav',
  width: '100%',
  display: 'flex',
  justifyContent: 'space-between',
});

export const APNavIcon = glamorous.div<{ disabled: boolean }>(
  {
    label: 'AP-nav-icon',
    border: `3px solid ${colours.$white}`,
    width: 35,
    height: 35,
    borderRadius: '50%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  ({ disabled }) => ({
    opacity: disabled ? 0.5 : 1.0,
  }),
);

// Poll Container

export const APNavHolder = glamorous.div({
  label: 'AP-nav-holder',
  width: 'calc(100% - 32px)',
  position: 'absolute',
  bottom: 110,
  left: 16,
  // right: 16,
  zIndex: 5,
  [mediaQueries.tabletPortrait]: {
    width: '100%',
    left: 0,
    bottom: 16,
    right: 'auto',
    padding: '0 12px',
  },
});

export const APProgressHolder = glamorous.div({
  label: 'AP-progress-holder',
  width: '100%',
  position: 'absolute',
  top: 8,
  left: 0,
  zIndex: 5,
  [mediaQueries.tabletPortrait]: {
    top: 12,
  },
});

export const APNavIntroHolder = glamorous(APNavHolder)({
  label: 'AP-nav-intro-holder',
  width: 'auto',
  right: 16,
  left: 'auto',
  bottom: 185,
  [mediaQueries.tabletPortrait]: {
    left: 'auto',
    right: 0,
    width: 105,
    display: 'flex',
    justifyContent: 'flex-end',
  },
});
