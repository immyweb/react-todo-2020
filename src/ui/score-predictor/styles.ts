import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { transparentize } from 'polished';
import { H3, H2 } from '../typography/typography';
import { fonts } from '../../styles/fonts';
import { mediaQueries } from '../../styles/breakpoints';

interface IGProps {
  fontName?: string;
  fontAlignment?: string;
  definedWidth?: string | number;
  definedHeight?: number;
  borderLeft?: boolean;
  borderRight?: boolean;
  width?: number;
  disabled?: boolean;
  draw?: boolean;
  lose?: boolean;
  marginBottom?: number;
  errorScreen?: boolean;
  fontsize?: number;
  fontbold?: boolean;
  align?: string;
  lineheight?: number;
  tableType?: string;
}

// Predictor
export const FixtureTable = glamorous.table({
  label: 'fixture-table',
  width: '100%',
  borderSpacing: 0,
});

export const THead = glamorous.thead({
  label: 'fixture-thead',
  background: colours.$offBlack,
  color: colours.$white,
  fontSize: 14,
  fontWeight: 'normal',
});

export const THeading = glamorous.th<IGProps>(
  {
    label: 'fixture-theading',
    padding: '5px 0',
    fontFamily: fonts.$Roboto,
    textAlign: 'center',
    position: 'relative',
  },
  ({ fontAlignment, borderLeft, borderRight, definedWidth }: any) => ({
    textAlign: fontAlignment,
    borderLeft: borderLeft ? `1px solid ${colours.$white}` : 'none',
    borderRight: borderRight ? `1px solid ${colours.$white}` : 'none',
    width: definedWidth,
  }),
);

export const THBorder = glamorous.span({
  label: 'fixture-th-border',
  position: 'absolute',
  right: 0,
  color: colours.$midGrey,
  fontFamily: fonts.$Roboto,
});

export const Badge = glamorous.img<IGProps>(
  {
    label: 'fixture-badge',
    height: 30,
    width: 'auto',
    display: 'block',
    margin: '0 auto',
  },
  ({ definedWidth, draw, lose }) => ({
    width: definedWidth,
    opacity: draw || lose ? 0.3 : 1,
  }),
);

export const BadgeLabel = glamorous.span<IGProps>(
  {
    label: 'fixture-badge-label',
  },
  ({ draw, lose }) => ({
    opacity: draw || lose ? 0.3 : 1,
  }),
);

export const TrTeam = glamorous.tr<IGProps>(
  {
    label: 'fixture-tr-team',
    backgroundColor: colours.$white,
    height: 58,
    borderBottom: `1px solid ${colours.$midLightGrey}`,
    '&:last-child': {
      borderBottom: 'none',
    },
  },
  ({ definedHeight }: any) => ({
    height: definedHeight,
  }),
);

export const TrPrem = glamorous(TrTeam)({
  '&:nth-child(-n+5)': {
    backgroundColor: colours.$lightGrey,
  },
  '&:nth-child(4)': {
    borderBottom: `1px dashed ${colours.$midGrey}`,
  },
  '&:nth-child(5)': {
    backgroundColor: colours.$ultraLightGrey,
    borderBottom: `1px dashed ${colours.$midGrey}`,
  },
  '&:first-child': {
    backgroundColor: colours.$midLightGrey,
  },
  '&:nth-last-child(-n+3)': {
    backgroundColor: colours.$lightGrey,
  },
  '&:nth-child(17)': {
    borderBottom: `1px dashed ${colours.$midGrey}`,
  },
  '&:last-child': {
    borderBottom: `3px solid ${colours.$darkGrey}`,
  },
});

export const TrChamp = glamorous(TrTeam)({
  '&:nth-child(-n+6)': {
    backgroundColor: colours.$lightGrey,
  },
  '&:nth-child(2)': {
    backgroundColor: colours.$midLightGrey,
    borderBottom: `1px dashed ${colours.$midGrey}`,
  },
  '&:nth-child(6)': {
    borderBottom: `1px dashed ${colours.$midGrey}`,
  },
  '&:first-child': {
    backgroundColor: colours.$midLightGrey,
    borderBottom: `1px solid ${colours.$midGrey}`,
  },
  '&:nth-last-child(-n+3)': {
    backgroundColor: colours.$lightGrey,
  },
  '&:nth-child(21)': {
    borderBottom: `1px dashed ${colours.$midGrey}`,
  },
  '&:last-child': {
    borderBottom: `3px solid ${colours.$darkGrey}`,
  },
});

export const TdTeam = glamorous.td<IGProps>(
  {
    label: 'fixture-td-team',
    textAlign: 'center',
    fontSize: 14,
    fontFamily: fonts.$theSunMedium,
    padding: '1px 0',
  },
  ({ fontAlignment, definedWidth, fontName }: any) => ({
    textAlign: fontAlignment,
    width: definedWidth,
    fontFamily: fontName,
  }),
);

export const TdCta = glamorous.td<IGProps>(
  {
    label: 'fixture-td-cta',
    textAlign: 'center',
  },
  ({ align }: any) => ({
    textAlign: align,
    [mediaQueries.tabletPortrait]: {
      textAlign: 'center',
    },
  }),
);

// Controls
export const NavHolder = glamorous.div({
  label: 'fixture-nav-holder',
  display: 'flex',
});

export const Nav = glamorous.button<IGProps>(
  {
    label: 'fixture-nav',
    textShadow: 'none',
    textDecoration: 'none',
    WebkitFontSmoothing: 'antialiased',
    background: 'none',
    display: 'inline-block',
    boxShadow: 'none',
    padding: 0,
    border: 'none',
    flexGrow: 1,
    ':focus': {
      outline: 0,
    },
  },
  ({ disabled }) => ({
    cursor: disabled ? 'default' : 'pointer',
  }),
);

export const PageIndex = glamorous.div({
  label: 'fixture-nav-index',
  textAlign: 'center',
  fontSize: 14,
  color: colours.$darkGrey,
  flexGrow: 2,
});

export const CurrentNum = glamorous.span({
  label: 'fixture-nav-index-current',
  color: colours.$football,
  fontWeight: 'bold',
});

export const SlideCount = glamorous.span({
  label: 'fixture-nav-index-total',
});

// Slides
export const SlidesContainer = glamorous.div({
  label: 'slides-container',
  width: '100%',
  overflow: 'hidden',
  borderBottom: `3px solid ${colours.$darkGrey}`,
  marginBottom: 15,
  position: 'relative',
});

// Confirmation and Error screen
export const MessageContainer = glamorous.section<IGProps>(
  {
    label: 'confirmation',
    background: transparentize(0.1, colours.$white),
    top: 0,
    textAlign: 'center',
    padding: 20,
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 10,
  },
  ({ width, errorScreen }) => ({
    width: width ? width : '100%',
    position: errorScreen ? 'relative' : 'absolute',
  }),
);

export const Heading = glamorous(H3)<IGProps>(
  {
    fontFamily: fonts.$theSunHeavyCondensed,
    textAlign: 'center',
    fontSize: 22,
  },
  ({ marginBottom }) => ({
    marginBottom: marginBottom ? marginBottom : 5,
  }),
);

export const Copy = glamorous.p<IGProps>(
  {
    fontFamily: fonts.$theSunMedium,
    textAlign: 'center',
    marginBottom: 10,
    marginTop: 10,
  },
  ({ fontsize, fontbold, lineheight }) => ({
    fontSize: fontsize ? fontsize : 16,
    fontWeight: fontbold ? 'bold' : 'normal',
    lineHeight: lineheight,
  }),
);

export const Graphic = glamorous.img({
  label: 'graphic',
  width: 68,
  height: 'auto',
  margin: '0 auto 10px auto',
});

// Final table

export const Headline = glamorous(H2)({
  label: 'final-table-heading',
  padding: '10px 20px',
  color: colours.$white,
  position: 'relative',
  background: colours.$football,
  marginBottom: 0,
});
