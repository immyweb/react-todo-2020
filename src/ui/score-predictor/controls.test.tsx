import * as React from 'react';
import { shallow } from 'enzyme';
import Controls from './controls';
import { CurrentNum, SlideCount, Nav } from './styles';

describe('<Controls />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Controls currentSlide={1} slideCount={6} onTogglePrev={() => {}} onToggleNext={() => {}} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct current slide number and total slide count', () => {
    const wrapper = shallow(
      <Controls currentSlide={3} slideCount={9} onTogglePrev={() => {}} onToggleNext={() => {}} />,
    );
    const currentSlide = wrapper.find(CurrentNum);
    expect(currentSlide.dive().text()).toEqual('4');
    const total = wrapper.find(SlideCount);
    expect(total.dive().text()).toEqual('9');
  });

  it('onTogglePrev is called when clicking back button', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <Controls currentSlide={3} slideCount={9} onTogglePrev={fn} onToggleNext={() => {}} />,
    );
    const btn = wrapper.find(Nav).first();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('onToggleNext is called when clicking forward button', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <Controls currentSlide={3} slideCount={9} onTogglePrev={() => {}} onToggleNext={fn} />,
    );
    const btn = wrapper.find(Nav).last();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('back button should be disabled if on first slide', () => {
    const wrapper = shallow(
      <Controls currentSlide={0} slideCount={6} onTogglePrev={() => {}} onToggleNext={() => {}} />,
    );
    const prevBtn = wrapper.find(Nav).first();
    expect(prevBtn.props().disabled).toEqual(true);
    const nextBtn = wrapper.find(Nav).last();
    expect(nextBtn.props().disabled).toEqual(false);
  });

  it('next button should be disabled if on last slide', () => {
    const wrapper = shallow(
      <Controls currentSlide={5} slideCount={6} onTogglePrev={() => {}} onToggleNext={() => {}} />,
    );
    const nextBtn = wrapper.find(Nav).last();
    expect(nextBtn.props().disabled).toEqual(true);
    const prevBtn = wrapper.find(Nav).first();
    expect(prevBtn.props().disabled).toEqual(false);
  });
});
