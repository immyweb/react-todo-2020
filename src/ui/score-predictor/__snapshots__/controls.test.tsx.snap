// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    currentSlide={1}
    onToggleNext={[Function]}
    onTogglePrev={[Function]}
    slideCount={6}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(button)
          disabled={false}
          onClick={[Function]}
        >
          <Icon
            iconColour="#222526"
            iconName="chevron-solid-left"
            iconSize={18}
          />
        </glamorous(button)>,
        <glamorous(div)>
          <glamorous(span)>
            2
          </glamorous(span)>
           /
           
          <glamorous(span)>
            6
          </glamorous(span)>
        </glamorous(div)>,
        <glamorous(button)
          disabled={false}
          onClick={[Function]}
        >
          <Icon
            iconColour="#222526"
            iconName="chevron-solid-right"
            iconSize={18}
          />
        </glamorous(button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            iconColour="#222526"
            iconName="chevron-solid-left"
            iconSize={18}
          />,
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "iconColour": "#222526",
            "iconName": "chevron-solid-left",
            "iconSize": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(span)>
              2
            </glamorous(span)>,
            " /",
            " ",
            <glamorous(span)>
              6
            </glamorous(span)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "2",
            },
            "ref": null,
            "rendered": "2",
            "type": [Function],
          },
          " /",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "6",
            },
            "ref": null,
            "rendered": "6",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            iconColour="#222526"
            iconName="chevron-solid-right"
            iconSize={18}
          />,
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "iconColour": "#222526",
            "iconName": "chevron-solid-right",
            "iconSize": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(button)
            disabled={false}
            onClick={[Function]}
          >
            <Icon
              iconColour="#222526"
              iconName="chevron-solid-left"
              iconSize={18}
            />
          </glamorous(button)>,
          <glamorous(div)>
            <glamorous(span)>
              2
            </glamorous(span)>
             /
             
            <glamorous(span)>
              6
            </glamorous(span)>
          </glamorous(div)>,
          <glamorous(button)
            disabled={false}
            onClick={[Function]}
          >
            <Icon
              iconColour="#222526"
              iconName="chevron-solid-right"
              iconSize={18}
            />
          </glamorous(button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Icon
              iconColour="#222526"
              iconName="chevron-solid-left"
              iconSize={18}
            />,
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "iconColour": "#222526",
              "iconName": "chevron-solid-left",
              "iconSize": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(span)>
                2
              </glamorous(span)>,
              " /",
              " ",
              <glamorous(span)>
                6
              </glamorous(span)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "2",
              },
              "ref": null,
              "rendered": "2",
              "type": [Function],
            },
            " /",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "6",
              },
              "ref": null,
              "rendered": "6",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Icon
              iconColour="#222526"
              iconName="chevron-solid-right"
              iconSize={18}
            />,
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "iconColour": "#222526",
              "iconName": "chevron-solid-right",
              "iconSize": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
